### ISSUE TITLE
cargo-lambda watch "No Child Processes" error

### ISSUE THREAD

**Issue by dastardlychimp** [cargo-lambda/cargo-lambda#347]
On osx x86, Ventura 13.1
cargo-lambda 0.16.1 (e0c2431 2023-01-26Z) installed via brew
rustc 1.67.0 (fc594f156 2023-01-24)

When I start watching with `cargo lambda watch` then invoke `cargo lambda invoke my-lambda --data-ascii '{}'` I get the error every few seconds:
```
ERROR internal error watching your project error=IoError { about: "waiting on process group", err: Os { code: 10, kind: Uncategorized, message: "No child processes" } }
```

Might be related to specifying a binary...

**Comment by dastardlychimp** [comment]
@calavera, two log files.
[logfiles.zip](https://github.com/cargo-lambda/cargo-lambda/files/10540567/logfiles.zip)

**Comment by relevantsam** [comment]
I am actively experiencing this issues. Log lines look similar but not available to share.

**Comment by relevantsam** [comment]
```
   Compiling regex v1.7.1
ERROR internal error watching your project error=IoError { about: "waiting on process group", err: Os { code: 10, kind: Uncategorized, message: "No child process" } }                                                                                                                                                                                                                                                 
   Compiling regex-syntax v0.6.28
   Compiling syn v1.0.107
   Compiling rustls v0.20.8
   Compiling regex v1.7.1
   Compiling regex-syntax v0.6.28      ] 164/230: rustls, syn, regex-syntax, regex                                                                                                                                                                                                                                                                                                                                     
   Compiling syn v1.0.107
   Compiling rustls v0.20.8
   Compiling regex v1.7.1
```

This usually is combined with spinning on compilation of this series of crates. `cargo lambda build` is fine, as a counter point.

**Comment by dastardlychimp** [comment]
If I run watch with the `no-reload` flag it is able to compile and I can invoke test events.

**Comment by greenwoodcm** [comment]
as discovered in [this issue](https://github.com/awslabs/aws-lambda-rust-runtime/issues/600), I think we need to add target directories for each of the examples to the example crates' `.gitignore` files.  otherwise cargo is mutating the target directory after `cargo lambda watch` starts up, which triggers re-compilation, which triggers killing the running service.

**Comment by greenwoodcm** [comment]
the target directories are already ignored in the root crate's `.gitignore` file.  it'd be nice to have `cargo lambda watch` respect that file.  but according to the [docs for the ignore-files crate](https://docs.rs/ignore-files/latest/ignore_files/fn.from_origin.html):

> Importantly, this should be called from the origin of the project, not a subfolder. This function will not discover the project origin, and will not traverse parent directories. Use the [project::origins](crate::project::origins) function for that.


that pointer to `project::origins` seems to be a dead link.  it'd be nice to have `cargo lambda watch` just find the upstream `.gitignore` files, which it sounds like should be possible given these docs.

**Comment by greenwoodcm** [comment]
#378 should address
